## Libraries

In [6]:
import pandas as pd

Just pandas library was used in this section.

## Spreadsheet

In [7]:
path = '../01_source/Rela‡Æo de expedientes para capturar arquivos no GED_resposta.xls'
data = pd.read_excel(path, sheet_name='REL. DOCUMENTOS COPIADOS')
data.head()

iSequencial GED         Expediente  iCodDocumento  \
0          6622932  03.000.00001/2017           5070   
1          6626319  16.000.00003/2017           5070   
2          6626401  16.000.00004/2017           5070   
3          6626489  11.000.00001/2017           5070   
4          6627051  04.000.00001/2017           5083   

                                     Nome do Arquivo  
0         03000000012017_6622932_inicial_lorena.pdf   
1                   16000000032017_6626319_p.i..pdf   
2                   16000000042017_6626401_p.i..pdf   
3   11000000012017_6626489_00122055920165030077.pdf   
4  04000000012017_6627051_NP00017480420165100002G...

Above was selected the second sheet ("REL. DOCUMENTOS COPIADOS") of the [provided worksheet](../01_source/Rela‡Æo%20de%20expedientes%20para%20capturar%20arquivos%20no%20GED_resposta.xls "") and transformed it into a pandas DataFrame.

In [8]:
data.shape

(2126, 4)

In [9]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2126 entries, 0 to 2125
Data columns (total 4 columns):
iSequencial GED    2126 non-null int64
Expediente         2126 non-null object
iCodDocumento      2126 non-null int64
Nome do Arquivo    2126 non-null object
dtypes: int64(2), object(2)
memory usage: 66.5+ KB


The DataFrame dimensions and its columns were studied. It contains 2126 lines and 4 columns: 'iSequencial GED', 'Expediente', 'iCodDocumento' and 'Nome do Arquivo'. The first and the third are int64 type, while the others are object type, and apparently no nulls values were identified.

## Data Exploration

In [12]:
data['iCodDocumento'].unique()

array([5070, 5083])

For the column 'iCodDocumento', it was analyzed for identification of unique values and only two values were found, having this in view it is possible to use it as a categorical variable and as a label for a prediction process. 

In [10]:
data['iSequencial GED'].nunique() == data.shape[0]

True

In [11]:
data['Expediente'].nunique() == data.shape[0]

False

In [13]:
data['Nome do Arquivo'].nunique() == data.shape[0]

True

In [14]:
data.groupby('Expediente').iCodDocumento.nunique().sort_values(ascending=False).head()

Expediente
11.003.00037/2017    2
02.000.00145/2017    2
10.000.00198/2017    2
04.000.00780/2017    2
10.000.00110/2017    2
Name: iCodDocumento, dtype: int64

The other three columns were tested as candidates to be an identification column. As shown above, it seems that 'iSequencial GED' and 'Nome do Arquivo' can be used as identification columns, whereas 'Expediente' appears to have duplicated values.

## Dataset

In [15]:
data.drop(['iSequencial GED', 'Expediente'], axis=1, inplace=True)
data.set_index(['Nome do Arquivo'], inplace=True)
data.head()

iCodDocumento
Nome do Arquivo                                                  
03000000012017_6622932_inicial_lorena.pdf                    5070
16000000032017_6626319_p.i..pdf                              5070
16000000042017_6626401_p.i..pdf                              5070
11000000012017_6626489_00122055920165030077.pdf              5070
04000000012017_6627051_NP00017480420165100002GR...           5083

In [16]:
data.to_csv('xls-dataset.csv', sep = ';')

In the end, 'Nome do Arquivo' was selected as a line indentification and converted into index. 'iSequencial GED' and 'Expediente' were dropped. The result DataFrame with 'Nome do Arquivo' as index and 'iCodDocumento' as label was saved into a [csv file](./xls-dataset.csv "xls dataset").